pkg/controller/operconfig/operconfig_controller.go

```diff
@@ -264,15 +268,22 @@ func (r *ReconcileOperConfig) Reconcile(ctx context.Context, request reconcile.R
 	}
 	objs = append([]*uns.Unstructured{app}, objs...)
 
-	// Set up the Pod reconciler before we start creating DaemonSets/Deployments
+	// Set up the Pod reconciler before we start creating DaemonSets/Deployments/StatefulSets
 	daemonSets := []types.NamespacedName{}
 	deployments := []types.NamespacedName{}
+	statefulSets := []types.NamespacedName{}
 	relatedObjects := []configv1.ObjectReference{}
 	for _, obj := range objs {
 		if obj.GetAPIVersion() == "apps/v1" && obj.GetKind() == "DaemonSet" {
 			daemonSets = append(daemonSets, types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()})
 		} else if obj.GetAPIVersion() == "apps/v1" && obj.GetKind() == "Deployment" {
 			deployments = append(deployments, types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()})
+		} else if obj.GetAPIVersion() == "apps/v1" && obj.GetKind() == "StatefulSet" {
+			statefulSets = append(statefulSets, types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()})
+			continue
+		} else if obj.GetClusterName() == cnoclient.ManagementClusterName {
```

Comment:
should that be changed to `if obj.GetClusterName() == cnoclient.ManagementClusterName {`?
If there is a daemonset or deployment for management cluster it won't hit that if

Reply:
you're right, daemonset/deployment also need to be added in relatedObjects.

Reply:
Fixed.
use separate `if` for management cluster objects.